### PR TITLE
Add package for mixins magefile targets

### PR DESCRIPTION
**Pull request by carolynvs**
# What does this change
The mage/mixins package contains common magefile targets that all mixins
use: build, test, publish, etc. This minimizes the amount of custom
logic in our magefiles and should make it easier to manage the magefile
in all the mixins going forward.

# What issue does it fix
Deals with some longstanding drift between all the mixins build script logic. This should get them all on the same page.

# Notes for the reviewer
I am testing this out with the docker-compose mixin first in https://github.com/getporter/docker-compose-mixin/pull/23. Please review these two PRs together and let me know if this looks like the right change to make to EVERY mixin repo. Once this is merged I'll make the same change everywhere, so I'd like to make sure this first PR has everything we want in it so I only have to update repos once. 😀 

# Checklist
- [ ] Unit Tests
- [ ] Documentation
- [ ] Schema (porter.yaml)

### CODE REVIEW

**Review by vdice — APPROVED** [review]
LGTM!

**Comment by carolynvs** [comment]
@vdice I tested out this branch on skeletor and the docker-compose mixin and made a few more changes. Would you please take a look at the new commits and let me know if they look ok?

I plan to update the remaining mixins after we merge this and tag a new prerelease.

File: mage/releases/publish.go [hunk 1]
```diff
@@ -72,17 +73,15 @@ exec echo "$GITHUB_TOKEN"
...
 }
 
 func publishPackage(pkgType string, name string) {
 	mg.Deps(tools.EnsureGitHubClient, ConfigureGitBot)
 
-	info := mage.LoadMetadata()
+	info := LoadMetadata()
 
-	repo := os.Getenv("PORTER_RELEASE_REPOSITORY")
+	repo := os.Getenv(ReleaseRepository)
```
Comment (vdice) [inline mage/releases/publish.go:84]: Wondering if we want to specifically error out if repo is empty (similarly for `publishPackageFeed` below and its `remote` value)?  Or, will it be obvious to the user that an empty value (via an unset env var) is to blame for this target not working?

**Review by vdice — APPROVED** [review]
Non-blocking Q/comment; LGTM still stands.

File: mage/mixins/magefile.go [hunk 2]
```diff
@@ -78,17 +78,14 @@ func (m Magefile) PublishMixinFeed() {
...
 func (m Magefile) TestPublish(username string) {
-	// Backup the git config file, publish will set the user to a bot
-	mgx.Must(shx.Copy(".git/config", ".git/config.bak"))
-
-	os.Setenv(releases.ReleaseRepository, fmt.Sprintf("github.com/%s/%s-mixin", username, m.MixinName))
-	os.Setenv(releases.PackagesRemote, fmt.Sprintf("https://github.com/%s/packages.git", username))
+	mixinRepo := fmt.Sprintf("github.com/%s/%s-mixin", username, m.MixinName)
+	pkgRepo := fmt.Sprintf("https://github.com/%s/packages.git", username)
+	fmt.Printf("Publishing a release to %s and committing a mixin feed to %s\n", mixinRepo, pkgRepo)
+	fmt.Printf("If you use different repository names, set %s and %s then call mage Publish instead.\n", releases.ReleaseRepository, releases.PackagesRemote)
```
Comment (vdice) [inline mage/mixins/magefile.go:84]: 👍

**Pull request merged**